crates/volta-core/src/tool/package/mod.rs

```diff
@@ -138,31 +138,37 @@ impl Display for Package {
 /// * the unpacked and initialized package
 pub fn uninstall(name: &str) -> Fallible<()> {
     let home = volta_home()?;
-    // if the package config file exists, use that to remove any installed bins and shims
     let package_config_file = home.default_package_config_file(name);
-    if package_config_file.exists() {
-        let package_config = PackageConfig::from_file(&package_config_file)?;
+    let package_binary_file = binaries_from_package(name)?;
 
-        for bin_name in package_config.bins {
-            remove_config_and_shim(&bin_name, name)?;
-        }
+    if package_config_file.exists() || !package_binary_file.is_empty() {
+        // if the package config file exists, use that to remove any installed bins and shims
+        if package_config_file.exists() {
+            let package_config = PackageConfig::from_file(&package_config_file)?;
 
-        fs::remove_file(&package_config_file)
-            .with_context(delete_file_error(&package_config_file))?;
-    } else {
-        // there is no package config - check for orphaned binaries
-        for bin_name in binaries_from_package(name)? {
-            remove_config_and_shim(&bin_name, name)?;
+            for bin_name in package_config.bins {
+                remove_config_and_shim(&bin_name, name)?;
+            }
+
+            fs::remove_file(&package_config_file)
+                .with_context(delete_file_error(&package_config_file))?;
+
+            info!("{} package '{}' uninstalled", success_prefix(), name);
+        } else {
+            // there is no package config - check for orphaned binaries
+            for bin_name in package_binary_file {
+                remove_config_and_shim(&bin_name, name)?;
+            }
         }
+    } else {
+        warn!("No package '{}' found to uninstall.", name);
```

Comment:
A suggestion for restructuring this a bit – since you end up repeating the checks inside the outer `if` expression, I'd just push this up to a single, three-scenario `if` expression:

```rust
if package_config_file.exists() {
    // if the package config file exists, use that to remove any installed bins and shims
    let package_config = PackageConfig::from_file(&package_config_file)?;

    for bin_name in package_config.bins {
        remove_config_and_shim(&bin_name, name)?;
    }

    fs::remove_file(&package_config_file)
        .with_context(delete_file_error(&package_config_file))?;

    info!("{} package '{}' uninstalled", success_prefix(), name);
} else if !package_binary_file.is_empty() {
    // there is no package config - check for orphaned binaries
    for bin_name in package_binary_file {
        remove_config_and_shim(&bin_name, name)?;
    }
} else {
    warn!("No package '{}' found to uninstall.", name);
}
```

Reply:
Thanks for the feedback. One question is that if we go with `if .. else if... else` as you suggested, if the code goes into the `else if` then we are not printing anything to the users right?

Reply:
Since I'm not quite sure if `!package_binary_file.is_empty()` part should be considered as a valid uninstall or just a clean up.

Reply:
I *think* Chuck covered this in his comments below, but if it's still unclear after working through that, I'll follow up!